src/Framework/Config/GacelaFileConfig/Factory/GacelaConfigUsingGacelaPhpFileFactory.php

```diff
@@ -37,13 +40,16 @@ public function createGacelaFileConfig(): GacelaConfigFileInterface
     {
         /** @var SetupGacelaInterface|callable $setupGacelaFile */
         $setupGacelaFile = $this->fileIo->include($this->gacelaPhpPath);
+
         if (is_callable($setupGacelaFile)) {
-            trigger_deprecation('Gacela', '0.15', 'Return a SetupGacelaInterface instance directly. The callable option will be removed in the next version.');
+            $gacelaConfig = new GacelaConfig($this->setup->externalServices());
+            $setupGacelaFile($gacelaConfig);
+            $setupGacela = SetupGacela::fromGacelaConfig($gacelaConfig);
+        } else {
+            $setupGacela = $setupGacelaFile;
```

Comment:
Will we drop in the future the `(SetupGacela())->setXXX()` construction?

Reply:
Good idea, but it will require some minor changes. Please, apply these changes in a follow PR.